### PR TITLE
fix: fix prog description display

### PR DESCRIPTION
**Pull request by feng-tao**
Signed-off-by: feng-tao <fengtao04@gmail.com>

### Summary of Changes

fix: https://github.com/amundsen-io/amundsen/issues/701

### Tests

_What tests did you add or modify and why? If no tests were added or modified, explain why. Remove this line_

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

**Comment by feng-tao** [comment]
@ttannis let me know.

**Comment by feng-tao** [comment]
many mypy error if package is setup correctly
```
mypy --ignore-missing-imports --follow-imports=skip --strict-optional --warn-no-return .
tests/unit/api/search/test_v0.py:84: error: Function is missing a type annotation for one or more arguments
tests/unit/api/search/test_v0.py:108: error: Function is missing a type annotation for one or more arguments
tests/unit/api/search/test_v0.py:140: error: Function is missing a type annotation for one or more arguments
tests/unit/api/search/test_v0.py:166: error: Function is missing a type annotation for one or more arguments
tests/unit/api/search/test_v0.py:323: error: "SearchUser" has no attribute "assertDictContainsSubset"
tests/unit/api/search/test_v0.py:422: error: Function is missing a type annotation for one or more arguments
tests/unit/api/search/test_v0.py:428: error: Need type annotation for 'test_filters'
tests/unit/api/search/test_v0.py:445: error: Function is missing a type annotation for one or more arguments
```

**Comment by feng-tao** [comment]
will create an issue and see if the community has anyone could led a hand

**Review by allisonsuarez — APPROVED** [review]
LGTM

**Review by dikshathakur3119 — APPROVED** [review]

File: amundsen_application/api/utils/metadata_utils.py [hunk 1]
```diff
@@ -183,9 +180,13 @@ def _convert_prog_descriptions(prog_descriptions: List = None) -> Dict:
                                 not in right_config, prog_descriptions))
             other.sort(key=lambda x: _sort_prog_descriptions(other_config, x))
 
-    updated_descriptions['left'] = left
-    updated_descriptions['right'] = right
-    updated_descriptions['other'] = other
+            updated_descriptions['left'] = left
```
Comment (ttannis) [inline amundsen_application/api/utils/metadata_utils.py:183]: I have to check some edge cases in the JS code, but can we add a unit test to cover `prog_display_config = None`?

Comment (feng-tao) [reply amundsen_application/api/utils/metadata_utils.py:183]: yeap, will do

Comment (ttannis) [reply amundsen_application/api/utils/metadata_utils.py:183]: I noticed that this implementation makes `left` and `right` optional on the dictionary, and verified that the JS code treats them optional as well so it shouldn't break anything.

However, when there is no configuration the behavior is just that the given `prog_descriptions` is simply not manipulated in any way. So what do you think of an alternative one line fix that just changes the initialization of `other` in the original implementation, so:
```
// line 160 of original implementation
other = prog_descriptions or [] # type: List
```

**Review by ttannis — COMMENTED** [review]
I think there's a simpler fix for this, but please let me know if there is something I'm missing.

File: amundsen_application/api/utils/metadata_utils.py (hunk 1 above)
Comment (feng-tao) [reply amundsen_application/api/utils/metadata_utils.py:183]: @ttannis https://github.com/amundsen-io/amundsenfrontendlibrary/pull/672 with a updated test as well
